src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -222,6 +221,43 @@ trait EntityComponent {
       entityAttributeQuery.batchInsertAttributes(attributeRecsToEntityId.keys.toSeq)
     }
 
+    private def upsertAttributes(entities: Traversable[Entity], entityRecs: Traversable[EntityRecord], referencedEntityRecs: Traversable[EntityRecord]) = {
+      def createEntityAttributeTempTable() = {
+        sql"""create temporary table if not exists ENTITY_ATTRIBUTE_TEMP (id bigint(20) unsigned NOT NULL AUTO_INCREMENT primary key, name text NOT NULL, value_string text, value_number double DEFAULT NULL, value_boolean bit(1) DEFAULT NULL, value_entity_ref bigint(20) unsigned DEFAULT NULL, list_index int(11) DEFAULT NULL, owner_id bigint(20) unsigned NOT NULL)""".as[Int]
+      }
+
+      def clearEntityAttributeTempTable() = {
+        sql"""delete from ENTITY_ATTRIBUTE_TEMP""".as[Int]
+      }
+
+      def insertTempAttributes(): WriteAction[Unit] = {
+        val entityIdsByName = referencedEntityRecs.map(r => AttributeEntityReference(r.entityType, r.name) -> r.id).toMap
+        val attributeRecsToEntityId = (for {
+          entity <- entities
+          (attributeName, attribute) <- entity.attributes
+          attributeRec <- entityAttributeQuery.marshalAttribute(entityIdsByName(entity.toReference), attributeName, attribute, entityIdsByName)
+        } yield attributeRec -> entityIdsByName(entity.toReference)).toMap
+
+        entityAttributeTempQuery.batchInsertAttributes(attributeRecsToEntityId.keys.toSeq)
+      }
+
+      object alterMasterActions extends RawSqlQuery {
+        val driver: JdbcDriver = EntityComponent.this.driver
+        val entityIds = entityRecs.map(_.id).mkString("(", ",", ")")
+        //if you drop the #$entityIds clause at the end it just may delete everything in the attributes table :)
+        val deleteFromMasterAction = sql"""delete ea from ENTITY_ATTRIBUTE ea left join ENTITY_ATTRIBUTE_TEMP ta on (ea.name,ea.owner_id,ea.list_index)<=>(ta.name,ta.owner_id,ta.list_index) where ta.owner_id is null and ea.owner_id in #$entityIds""".as[Int]
```

Comment:
one problem with my query is that it will not work when all attributes of an entity are being deleted because the owner id will not exist in ENTITY_ATTRIBUTE_TEMP